java/test/jmri/swing/PreferencesPanelTestBase.java

```diff
@@ -0,0 +1,99 @@
+package jmri.swing;
+
+import jmri.util.JUnitUtil;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.assertj.core.api.Java6Assertions.catchThrowable;
+
+/**
+ *  Base Tests for implementations of the PreferenesPanel interface.
+ *
+ * @author Paul Bender Colyright (C) 2020
+ */
+abstract public class PreferencesPanelTestBase {
```

Comment:
This class should use a generic `PreferencesPanelTestBase<P extends PreferencesPanel>` to avoid any potential need to cast `prefsPanel` to the tested class in the actual test classes.

Reply:
thanks. will fix